minigames/treeballz.py

```diff
@@ -0,0 +1,139 @@
+            elif(i == 0):
+                self.screen.blit(TreeBallz.playerImages[0], eyeball)
+            else:
+                self.screen.blit(TreeBallz.playerImages[2], eyeball)
+        for i,list in enumerate(self.projectiles):
+            for proj in list:
+                self.screen.blit(TreeBallz.projectileImages[i],proj[0]['rect'])
+
+    def addProjectile(self,i,lookingLeft):
+        self.projectiles[i].append([{'rect': pygame.Rect(self.eyeballs[i].x,self.eyeballs[i].y+5, 16, 16), 'leftFacing': lookingLeft}])
+
+
+    def get_duration(self):
+        return 10000
+
+    def update(self):
+        pygame.event.get()
+
+        #Ladder Collision
+        for i in range(2):
+            ladderHits = 0
+            for ladder in self.ladders:
+                if self.eyeballs[i].colliderect(ladder):
+                    self.inLadder[i] = True
+                    ladderHits += 1
+            if ladderHits == 0:
+                self.inLadder[i] = False
+
+        #Shitty gravity
+        for i,eye in enumerate(self.eyeballs):
+            onPlatform = False
+            for platform in self.platforms:
+                if eye.colliderect(platform) and not self.inLadder[i]:# and eye.bottom <= platform.bottom - platform.height/3:
+                    eye.bottom = platform.top + 1 
+                    onPlatform = True
+            print onPlatform
+            if eye.bottom < 480 and not self.inLadder[i] and not onPlatform:
+                eye.y += 15
+            elif eye.bottom > 480:
+                eye.bottom = 480
+
+        #Player movement and shooting
+        for i in range(2):
+            keys = input_map.get_player_keys(i)
+            if len(keys) > 0 and not self.dead[i]:
+                if keys[input_map.RIGHT]:
+                    if self.eyeballs[i].right < 725:
+                        self.eyeballs[i].x += 10
+                        self.lookingLeft[i] = False
+                if keys[input_map.LEFT]:
+                    if self.eyeballs[i].left > 25:
+                        self.eyeballs[i].x -= 10
+                        self.lookingLeft[i] = True
+                if keys[input_map.ACTION]:
+                    if self.lastShot[i] > 80:
+                        self.addProjectile(i,self.lookingLeft[i]);
+                        self.lastShot[i] = 0.0
+                if keys[input_map.UP] and self.inLadder[i]:
+                        self.eyeballs[i].y -= 10
+                if keys[input_map.DOWN] and self.inLadder[i]:
+                        self.eyeballs[i].y += 10
+
+        #TODO: Projectile movement.
+        for i,list in enumerate(self.projectiles):
+            for proj in list:
+                if proj[0]['leftFacing'] == True:
+                    proj[0]['rect'].x -= 20
+                else:
+                    proj[0]['rect'].x += 20
+
+                if i==0 and proj[0]['rect'].colliderect(self.eyeballs[1]):
+                    self.dead[1] = True
+                    list.remove(proj)
```

Comment:
Je sais qu'on s'en sacre parce que tu finis la game tout de suite après, mais j'ai toujours des frissons quand je vois quelqu'un enlever des items d'une liste qu'il itère. (http://stackoverflow.com/questions/4960968/how-to-safely-remove-elements-from-a-list-in-python)